uwblueprint/supportive-housing
# Create tags route

## Notion task link


[Create tags route](https://www.notion.so/uwblueprintexecs/Create-Tags-Route-7f4e8763fdd1437d8954ac42e918c886)



## Implementation description
* Updated tags model to add a unique constraint to name
* Added create route
* Modified update route to catch non-unique name error



## Steps to test
1. Upgrade database with latest migration: `bash ./scripts/flask-db-upgrade.sh`
2. In order to test these routes with Postman, first comment out `@require_authorization_by_role({"Admin"})` on lines 47 and 71 of backend/app/rest/tags_routes.py.
3. Create: Create a tag by POSTing to /tags/ (e.g. {"name": “Cool Tag”, "status": "Active”}). This should show up in your database. Try creating another tag with the same name - you should get an error telling you that name already exists.
4. Update: Update tag by id by PUTing to /tags/[id] an updated tag (e.g. {"name": “Epic Tag”, "status": "Active”}). Database should show modified tag. Try updating a different tag to have the same name - you should get an error telling you that name already exists. Also try updating a non-existent id - you should get an error telling you that id was not found.



## What should reviewers focus on?
* 


## Checklist
